kratos/solving_strategies/schemes/residual_based_bdf_custom_scheme.h

```diff
@@ -94,33 +94,43 @@ class ResidualBasedBDFCustomScheme
     ///@name Life Cycle
     ///@{
 
+    /**
+     * @brief Constructor. The BDF method
+     * @param ThisParameters The parameters containing the list of variables to consider
+     * @todo The ideal would be to use directly the dof or the variable itself to identify the type of variable and is derivatives
+     */
+    explicit ResidualBasedBDFCustomScheme(Parameters ThisParameters)
+        : ResidualBasedBDFCustomScheme(ThisParameters.Has("integration_order") ? static_cast<std::size_t>(ThisParameters["integration_order"].GetInt()) : 2, ThisParameters)
+    {
+    }
+
     /**
      * @brief Constructor. The BDF method
      * @param Order The integration order
-     * @param rParameters The parameters containing the list of variables to consider
+     * @param ThisParameters The parameters containing the list of variables to consider
      * @todo The ideal would be to use directly the dof or the variable itself to identify the type of variable and is derivatives
      */
     explicit ResidualBasedBDFCustomScheme(
         const std::size_t Order = 2,
-        Parameters rParameters =  Parameters(R"({})")
+        Parameters ThisParameters =  Parameters(R"({})")
```

Comment:
for example, why do you have this constructor at all? 

if you have the one with parameters, than you could do 

     explicit ResidualBasedBDFCustomScheme(
             Parameters rParameters =  Parameters(R"({})")	       
     {
       //GetDefaults shall be implemented in this class, eventually overwriting the base one
      Parameters default_parameters = GetDefaultParameters();
      ThisParameters.ValidateAndAssignDefaults(default_parameters);

       //now here call the base class constructor
       BDFBaseType( ThisParameters["order"].GetInt())
    }

note that an important advantage of doing this is that when you print the parameters you pass, they will reflect what you are actually using, while with your version the parameters that you are using would not be stored in the "ThisParameters"

note also that if the default parameters of the derived class is a superset of the one of the base class, then in the derived GetDefaultParameters you can directly use internally the ones in the base class and merge to it ... but you could also change the defaults if you like.

Reply:
This constructor is in the master, is the current default constructor, so it will not go away at least now

Reply:
I tried this 

~~~
BDFBaseType( ThisParameters["order"].GetInt())
~~~

And it didn't work ... :(, will try again